localcmd/local.go

```diff
@@ -17,28 +18,51 @@ type Format string
 
 const (
 	FormatCSV  Format = "csv"
-	FormatJSON        = "json"
+	FormatJSON Format = "json"
 )
 
 type Args struct {
 	Start      time.Time
+	Format     Format
 	MetricStat *types.MetricStat
 }
 
-type nopMetricStore struct {
-}
+type nopMetricStore struct{}
 
-func (_ nopMetricStore) Get(ctx context.Context, m *types.MetricStat) (lastStart time.Time, ok bool, err error) {
+func (nopMetricStore) Get(ctx context.Context, m *types.MetricStat) (lastStart time.Time, ok bool, err error) {
 	return
 }
-func (_ nopMetricStore) Put(ctx context.Context, m *types.MetricStat, lastStart time.Time) (err error) {
+func (nopMetricStore) Put(ctx context.Context, m *types.MetricStat, lastStart time.Time) (err error) {
 	return
 }
 
 type csvPutter struct {
 	writer csv.Writer
 }
 
+type jsonPutter struct {
+	encoder *json.Encoder
+}
+
+func NewJSONPutter() jsonPutter {
+	return jsonPutter{
+		encoder: json.NewEncoder(os.Stdout),
+	}
+}
+
+func NewCSVPutter() csvPutter {
+	return csvPutter{
+		writer: *csv.NewWriter(os.Stdout),
+	}
+}
```

Comment:
Actually realised afterwards that I could have just encapsulated this in a single function that takes a format and returns a putter (as apposed to having to construct the putter yourself)